coalib/results/SourceRange.py

```diff
@@ -120,6 +120,43 @@ def expand(self, file_contents):
                                        tr.end.line,
                                        tr.end.column)
 
+    @enforce_signature
+    def affected_source(self, file_dict: dict):
+        r"""
+        Make an iterator that tells which are the lines affected in the
+        file within a specified range.
+
+        >>> from os.path import abspath
+        >>> sr = SourceRange.from_values('file_name', start_line=2, end_line=2)
+        >>> sr.affected_source({abspath('file_name'):
+        ...     ('def fun():\n', '    x = 2  \n')})
+        ('    x = 2  \n',)
+
+        If more than one line is affected.
+
+        >>> sr = SourceRange.from_values('file_name', start_line=2, end_line=3)
+        >>> sr.affected_source({abspath('file_name'):
+        ...     ('def fun():\n', '    x = 2  \n', '    print(x)  \n')})
+        ('    x = 2  \n', '    print(x)  \n')
+
+        :param file_dict:
+            It is a dictionary where the file names are the keys and
+            the contents of the files are the values(which is of type tuple).
+        :raises KeyError:
+            Raised when the file name is not present in ``file_dict``.
+        :return:
+            An iterator that returns affected lines in the specified file.
+            If the file is not affected return ``None``
+        """
+        try:
+            file_contents = file_dict[self.start.file]
+            if self.start.line and self.end.line:
+                # line number starts from 1, index starts from 0
+                return file_contents[self.start.line - 1: self.end.line]
+        except KeyError:
```

Comment:
Uh, wouldn't a simple `if self.start.file in file_dict` be better than a try-except? try-excepts are computationally costlier and we should try to be as performant as possible (plus a `if` is more readable anyway).